src/webnn_native/dml/GraphDML.cpp

```diff
@@ -460,11 +460,22 @@ namespace webnn_native { namespace dml {
         return std::to_string(type);
     }
 
+    std::string Graph::GetDeviceName() {
+        ml::DevicePreference devicePreference = mContextOptions.devicePreference;
+        if (devicePreference == ml::DevicePreference::Gpu ||
+            devicePreference == ml::DevicePreference::Default) {
+            return "GPU";
+        } else {
+            return "CPU";
+        }
+    }
+
     Graph::Graph(Context* context) : GraphBase(context) {
+        bool useGpu = GetDeviceName() == "GPU";
 #if defined(_DEBUG)
-        mDevice.reset(new ::pydml::Device(true, true));
+        mDevice.reset(new ::pydml::Device(useGpu, true));
```

Comment:
For DirectML, I suppose the default should be GPU.

Reply:
> For DirectML, I suppose the default should be GPU.

Yes, I already defined as this. Thanks! @huningxin

Reply:
@wchao1115, this PR maps [`MLDevicePreference.cpu`](https://webmachinelearning.github.io/webnn/#dom-mldevicepreference-cpu) to WRAP device of DirectML backend. What do you think?

Reply:
@huningxin DirectML works with WARP device just fine if that's what you ask. The pydml wrapper as of today ignores the power hint and always select the default GPU adapter. For completeness, in the future this might need to change to consider the power preference.

Reply:
Thanks @wchao1115. The power preference will be supported later.